Source/Core/Common/OpenXR.cpp

```diff
@@ -0,0 +1,714 @@
+    {
+      s_enabled_extensions.push_back(name);
+      return true;
+    }
+
+    return false;
+  };
+
+  // We'll initialize OpenXR with all available potentially used extensions.
+  TryEnableExtension("XR_MND_headless");
+  TryEnableExtension("XR_KHR_opengl_enable");
+  TryEnableExtension("XR_KHR_vulkan_enable");
+
+#ifdef WIN32
+  TryEnableExtension("XR_KHR_D3D11_enable");
+  TryEnableExtension("XR_KHR_D3D12_enable");
+#endif
+
+  XrInstanceCreateInfo info{XR_TYPE_INSTANCE_CREATE_INFO};
+  info.enabledExtensionNames = s_enabled_extensions.data();
+  info.enabledExtensionCount = uint32_t(s_enabled_extensions.size());
+
+  std::strcpy(info.applicationInfo.applicationName, "dolphin-emu");
+  info.applicationInfo.applicationVersion = 1;
+  std::strcpy(info.applicationInfo.engineName, "");
+  info.applicationInfo.engineVersion = 0;
+  info.applicationInfo.apiVersion = XR_CURRENT_API_VERSION;
+
+  XrResult result = xrCreateInstance(&info, &s_instance);
+  if (XR_FAILED(result))
+  {
+    ERROR_LOG(VIDEO, "OpenXR: xrCreateInstance: %d", result);
+    return false;
+  }
+
+  XrInstanceProperties instance_properties{XR_TYPE_INSTANCE_PROPERTIES};
+  result = xrGetInstanceProperties(s_instance, &instance_properties);
+  if (XR_FAILED(result))
+  {
+    WARN_LOG(VIDEO, "OpenXR: xrGetInstanceProperties: %d", result);
+  }
+  else
+  {
+    INFO_LOG(VIDEO, "OpenXR: Runtime name: %s", instance_properties.runtimeName);
+  }
+
+  XrSystemGetInfo system_get_info{XR_TYPE_SYSTEM_GET_INFO};
+  system_get_info.formFactor = XR_FORM_FACTOR_HEAD_MOUNTED_DISPLAY;
+  result = xrGetSystem(s_instance, &system_get_info, &s_system_id);
+  if (XR_FAILED(result))
+  {
+    ERROR_LOG(VIDEO, "OpenXR: xrGetSystem: %d", result);
+
+    Shutdown();
+
+    return false;
+  }
+
+  XrSystemProperties sys_props = {XR_TYPE_SYSTEM_PROPERTIES};
+  result = xrGetSystemProperties(s_instance, s_system_id, &sys_props);
+  if (XR_FAILED(result))
+  {
+    WARN_LOG(VIDEO, "OpenXR: xrGetSystemProperties: %d", result);
+  }
+  else
+  {
+    INFO_LOG(VIDEO, "OpenXR: System name: %s", sys_props.systemName);
+  }
+
+  s_run_event_thread = true;
+  s_event_thread = std::thread(EventThreadFunc);
+
+  return true;
+}
+
+std::unique_ptr<Session> CreateSession(const std::vector<std::string_view>& required_extensions,
+                                       const void* graphics_binding,
+                                       const std::vector<int64_t>& swapchain_formats)
+{
+  auto session = std::make_unique<Session>();
+  if (session->Create(required_extensions, graphics_binding) &&
+      session->CreateSwapchain(swapchain_formats))
+  {
+    return session;
+  }
+  else
+  {
+    return {};
+  }
+}
+
+Session::~Session()
+{
+  Destroy();
+}
+
+bool Session::Create(const std::vector<std::string_view>& required_extensions,
+                     const void* graphics_binding)
+{
+  Init();
+
+  if (!IsInit())
+    return false;
+
+  // If a session has already been created.
```

Comment:
This comment feels like you stopped writing in the middle (even though it ends in a full-stop). Would you mind rewriting it a little to indicate _why_ only one session is supported at the same time (well, duh, guessing because everything is `static` up there; but this sentence alone leaves more questions than answers)

Reply:
I'll make the comment more clear. It doesn't involve the `static`s.